test/AutoDiff/simple_aggregates.swift

```diff
@@ -0,0 +1,56 @@
+// RUN: %target-run-simple-swift
+// REQUIRES: executable_test
+
+import StdlibUnittest
+
+var SimpleAggregateTests = TestSuite("SimpleAggregate")
+
+struct DenseLayer : Equatable {
+  let w: Float
+  let b: Float
+}
+
+// TODO: Deduplicate these when it's possible.
+func prediction1(_ model: DenseLayer, _ input: Float) -> Float {
+  return input * model.w + model.b
+}
+func prediction2(_ model: DenseLayer, _ input: Float) -> Float {
+  return input * model.w + model.b
+}
+func prediction3(_ model: DenseLayer, _ input: Float) -> Float {
+  return input * model.w + model.b
+}
+
+struct Model : Equatable {
```

Comment:
Why are these tests relevant? They are loadable, so they won't call into `materializeAdjointIndirect` anyways if I remember correctly.

Reply:
It seems to be calling `materializeIndirectAggregate in [this place](https://github.com/apple/swift/blob/cd12cd45962c981ebec9fe149acb3eb327894f41/lib/SILOptimizer/Mandatory/TFDifferentiation.cpp#L3546) that doesn't know how to do stuff directly yet.

I expect we can teach that place to do things directly, and once we do that, this test will stop exercising the code path that I added in this PR. This test still seems generally good to have, as an integration test to make sure that various common model things work.

I couldn't figure out a way to add a test that exercises this PR's code path more reliably, because all the non-loadable values that I create crash AD for one reason or another. We can probably add a test that exercises this PR's code path as a part of generic AD work.

Reply:
Created https://bugs.swift.org/browse/SR-8769